src/main/scala/org/clulab/habitus/utils/ContextExtractor.scala

```diff
@@ -0,0 +1,260 @@
+package org.clulab.habitus.utils
+
+import org.clulab.habitus.variables.EntityDistFreq
+
+import org.clulab.odin.{Attachment, Mention}
+import org.clulab.processors.Document
+
+import scala.collection.mutable
+import scala.collection.mutable.ArrayBuffer
+
+trait Context extends Attachment {
+  def getArgValuePairs() = this.getClass.getDeclaredFields.toList
+    .map(arg => {
+      arg.setAccessible(true)
+      (arg.getName, arg.get(this))
+    })
+
+  def getTSVContextHeader() = {
+    getArgValuePairs().map(_._1).mkString("\t")
+  }
+
+  def getTSVContextString() = {
+    getArgValuePairs().map(_._2).mkString("\t")
+  }
+}
+
+case class DefaultContext(location: String, date: String, crop: String, fertilizer: String, comparative: Int) extends Context
+
+trait ContextExtractor {
+
+  def getContextPerMention(mentions: Seq[Mention], entityHistogram: Seq[EntityDistFreq], doc: Document, label: String): Seq[Mention]
+
+  def getComparative(mention: Mention): Int = {
+    val relative = Seq("vs", "vs.", "respectively")
+    if (mention.sentenceObj.words.intersect(relative).nonEmpty) 1 else 0
```

Comment:
If this runs lots of times and efficiency is important, instead of calculating the exact intersection, one might look for a single match and stop with
```scala
relative.exists(mention.sentenceObj.words.contains(_))
```
It's not as pretty, granted.  I guess that the Seq could be a simple Array and it could be stored statically in a companion object somewhere, in case it is being recreated for each call.